### DISCUSSION TITLE
[FEATURE]: Enable the ability to reattempt a workflow with the option to modify input parameters.

### DISCUSSION THREAD

**Discussion by MindfulAditya** [conductor-oss/conductor#74] [General]
In the current Netflix conductor workflows, the "Retry - from failed task" option is available. However, it lacks the capability to adjust input parameters during the retry process.

Scenario: Imagine a workflow with multiple tasks, for example, 10 tasks. If the 7th task encounters a failure due to incorrect input parameters, there is a need to restart the workflow from the 7th task, allowing the modification of input parameters for a successful rerun.

Please inform me whether this feature is currently accessible or if there are any workarounds in place.

I started this discussion thread before opening a feature request.

**Comment by rajanarahul93** [comment]
> In the current Netflix conductor workflows, the "Retry - from failed task" option is available. However, it lacks the capability to adjust input parameters during the retry process.
> 
> Scenario: Imagine a workflow with multiple tasks, for example, 10 tasks. If the 7th task encounters a failure due to incorrect input parameters, there is a need to restart the workflow from the 7th task, allowing the modification of input parameters for a successful rerun.
> 
> Please inform me whether this feature is currently accessible or if there are any workarounds in place.
> 
> I started this discussion thread before opening a feature request.

Hey there!

It seems like the "Retry - from failed task" option in Netflix Conductor workflows doesn't currently allow tweaking input parameters during the retry. Picture this: you've got a workflow with a bunch of tasks, and let's say the 7th task fails because of wonky input parameters. Now, the catch is, when you hit that retry button, there's no easy way to fix those inputs for a successful do-over.

You've started this discussion thread, which is a smart move before diving into a feature request. It's a chance for the Conductor community to weigh in – who knows, someone might have a workaround up their sleeve.

As for workarounds, it might involve some creative workflow design, maybe introducing conditions or an extra step for adjusting parameters before hitting that troublesome task.

For the latest scoop, I'd recommend checking out the freshest Conductor documentation or jumping into the discussion forums. And, of course, if you decide to fire off that feature request, make sure to paint a clear picture of why it's essential – real-world examples always help.

**Reply by VerstraeteBert** [reply]
What in the ChatGPT

**Comment by VerstraeteBert** [comment]
Hi @MindfulAditya ! I've recently implemented a similar feature in my company fork of Conductor. It's limited to changing the input of workflows and subworkflows. In essence, it allows to pass in a Map<String, Object> to the /retry endpoint, and this will be merged with the existing workflow inputs. I.e., overlapping keys have their values replaced, non-overlapping keys are both kept. If this is already enough for your use case, let me know. Could potentially also expand this to be possible on both the workflow and task level.

**Comment by JCHacking** [comment]
There is an option in the api called _rerun_, I recommend you to take a look at it in case it is useful for this.

**Comment by c4lm** [comment]
Closing since /rerun was mentioned.
